analyzers/number_function_argument_analyzer.go

```diff
@@ -33,10 +31,11 @@ func ReplacementHint(
 	r ast.Range,
 ) *analysis.Diagnostic {
 	return &analysis.Diagnostic{
-		Location: location,
-		Range:    r,
-		Category: "lint",
-		Message:  fmt.Sprintf("consider replacing with: `%s`", expr),
+		Location:         location,
+		Range:            r,
+		Category:         "replacement-hint",
```

Comment:
What are all possible lint categories? Does each type of hint belong to its own category?

Perhaps define enum-like constants for 'category'.